LuaUI/Widgets/unit_starlight_targetting.lua

```diff
@@ -0,0 +1,172 @@
+function widget:GetInfo()
+   return {
+      name         = "Starlight Targetting",
+      desc         = "Starlight targetting 0.01",
+      author       = "dyth68",
+      date         = "2022",
+      license      = "PD", -- should be compatible with Spring
+      layer        = 11,
+	  handler		= true, --for adding customCommand into UI
+      enabled      = true
+   }
+end
+
+
+local UPDATE_FRAME=2
+local StarlightStack = {}
+local GetUnitPosition = Spring.GetUnitPosition
+local GiveOrderToUnit = Spring.GiveOrderToUnit
+local GetUnitsInCylinder = Spring.GetUnitsInCylinder
+local GetUnitIsDead = Spring.GetUnitIsDead
+local GetMyTeamID = Spring.GetMyTeamID
+local GetUnitDefID = Spring.GetUnitDefID
+local GetTeamUnits = Spring.GetTeamUnits
+local GetUnitStates = Spring.GetUnitStates
+local Echo = Spring.Echo
+local GetSpecState = Spring.GetSpectatingState
+local CMD_STOP = CMD.STOP
+local CMD_ATTACK = CMD.ATTACK
+local CMD_UNIT_SET_TARGET = 34923
+
+local Starlight_ID = UnitDefNames.mahlazer.id
+local StarlightUnitDefID = UnitDefNames["mahlazer"].id
+
+local immobiles = {}
+for unitDefID, unitDef in pairs(UnitDefs) do
+  if unitDef.isImmobile then
+    immobiles[unitDefID] = true
+  end
+end
+
+function sq(x)
+	return x*x
+end
+
+function sqdist(p, q)
+	return sq(p[1]-q[1]) + sq(p[2]-q[2]) + sq(p[3]-q[3])
+end
+
+local StarlightControllerMT
+local StarlightController = {
+	unitID,
+	pos,
+	targetPos,
+	currentTarget,
```

Comment:
comment out (atm it's referencing some random nil globals)
```suggestion
	-- unitID, pos, targetPos, currentTarget
```